infra/cifuzz/fuzz_target.py

```diff
@@ -111,68 +113,57 @@ def _download_corpus(self):
-
-  def fuzz(self, use_corpus=True, extra_libfuzzer_options=None):
+    with clusterfuzz.environment.Environment(config_utils.DEFAULT_ENGINE,
+                                             self.config.sanitizer,
+                                             self.target_path,
+                                             interactive=True):
+      engine_impl = clusterfuzz.fuzz.get_engine(config_utils.DEFAULT_ENGINE)
+      result = engine_impl.minimize_corpus(self.target_path, [],
+                                           [self.latest_corpus_path],
+                                           self.pruned_corpus_path,
+                                           self.workspace.artifacts,
+                                           self.duration)
+
+    return FuzzResult(None, result.logs, self.pruned_corpus_path)
+
+  def fuzz(self):
     """Starts the fuzz target run for the length of time specified by duration.
 
     Returns:
       FuzzResult namedtuple with stacktrace and testcase if applicable.
     """
     logging.info('Running fuzzer: %s.', self.target_name)
-    if extra_libfuzzer_options is None:
-      extra_libfuzzer_options = []
-    env = base_runner_utils.get_env(self.config, self.workspace)
-    # TODO(metzman): Is this needed?
-    env['RUN_FUZZER_MODE'] = 'interactive'
-
-    if use_corpus:
-      # If corpus can be downloaded, use it for fuzzing.
-      self._download_corpus()
-      env['CORPUS_DIR'] = self.latest_corpus_path
-
-    options = LIBFUZZER_OPTIONS.copy() + [
-        f'-max_total_time={self.duration}',
-        # Make sure libFuzzer artifact files don't pollute $OUT.
-        f'-artifact_prefix={self.workspace.artifacts}/'
-    ] + extra_libfuzzer_options
-    command = ['run_fuzzer', self.target_name] + options
-
-    logging.info('Running command: %s', command)
-    process = subprocess.Popen(command,
-                               env=env,
-                               stdout=subprocess.PIPE,
-                               stderr=subprocess.PIPE)
 
-    try:
-      _, stderr = process.communicate(timeout=self.duration + BUFFER_TIME)
-    except subprocess.TimeoutExpired:
-      logging.error('Fuzzer %s timed out, ending fuzzing.', self.target_name)
-      return FuzzResult(None, None, self.latest_corpus_path)
+    self._download_corpus()
+    corpus_path = self.latest_corpus_path
+
+    logging.info('Starting fuzzing')
+    with clusterfuzz.environment.Environment(config_utils.DEFAULT_ENGINE,
+                                             self.config.sanitizer,
+                                             self.target_path,
+                                             interactive=True) as env:
+      engine_impl = clusterfuzz.fuzz.get_engine(config_utils.DEFAULT_ENGINE)
+      options = engine_impl.prepare(corpus_path, env.target_path, env.build_dir)
+      options.merge_back_new_testcases = False
+      options.analyze_dictionary = False
+
+      result = engine_impl.fuzz(self.target_path, options,
```

Comment:
How can we pass options for len_control=0 and -seed?
len_control is probably a good idea because the fuzz targets in CIFuzz don't have forever to grow the length like they do on ClusterFuzz.
-seed is important because we want CI fuzzing runs to be deterministic.

Also, is libClusterFuzz's selection of strategies deterministic?

Reply:
Done! This can just be appended to `options.arguments`. 

Strategies are completely disabled for libClusterFuzz.